app/routes.py

```diff
@@ -1 +1,239 @@
-from flask import Blueprint
\ No newline at end of file
+import os
+from datetime import datetime
+import requests
+from app import db
+
+from flask import Blueprint, jsonify, make_response, request, abort
+
+from app.models.task import Task
+from app.models.goal import Goal
+
+
+# DEFINING THE BLUEPRINTS
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+# error handling
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"message":f"task {model_id} invalid"}, 400))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"details": f"{cls.__name__} id {model_id} not found"}, 404))
+
+    return model
+
+
+# ============================= CRUD ROUTES  =============================
+# CREATE TASK- POST request to /tasks
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+    
+    if "title" not in request_body or "description" not in request_body:
+        return {"details": "Invalid data"}, 400
+    
+    new_task = Task.from_dict(request_body)
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return new_task.to_dic(), 201
+    
+
+# READ ALL TASKS- GET request to /tasks
+@tasks_bp.route("", methods=["GET"])
+def read_all_tasks():
+    sorted_by_query = request.args.get("sort")
+    title_query = request.args.get("title")
+    
+    if sorted_by_query== "asc":
+        tasks = Task.query.order_by(Task.title).all()
+    elif sorted_by_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc()).all()
+    elif sorted_by_query == "id":
+        tasks = Task.query.order_by(Task.task_id).all()
+    elif title_query:
+        tasks = Task.query.filter_by(title=title_query).all()
+    
+    if not sorted_by_query:  
+        tasks= Task.query.all()
+    
+    task_response = [task.to_dic()["task"] for task in tasks]
+    return jsonify(task_response), 200
+
+
+# READ ONE TASK- GET request to /tasks/<task_id>
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id):
+    task = validate_model(Task, task_id)
+    return task.to_dic(), 200
+
+
+# UPDATE TASK- PUT request to /tasks/<task_id>
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+    request_body = request.get_json()
+    
+    task.title = request_body["title"]
+    task.description = request_body["description"]
```

Comment:
Right now, if a user sends a request without the keys title or description your server would crash. There's a couple of ways to handle this, you could call the `validate_request` function before you access the keys in `request_body` or you could implement a `try/except` block.